src/socketHandler/gameHandler.js

```diff
@@ -1,34 +1,108 @@
-const { startRound } = require('../services/game.service');
-const { createGame } = require('../services/gamestate.service');
+const {
+  createGame,
+  getRounds,
+  addCardPlayed,
+  setStichPlayer,
+  setNextPlayer,
+  addRound,
+  addSubround,
+  getPlayers,
+  addPrediction,
+  getPredictionsForCurrentRound, getPredictionCount,
+  getNextPlayer,
+} = require('../services/gamestate.service');
+const { startRound, getWinningCard } = require('../services/game.service');
 const { getCurrentLobby } = require('../services/lobby.service');
 const { getByWebsocket } = require('../services/user.service');
+const Card = require('../utils/card.model');
 
 const startGame = async function (socket, io) {
   const user = await getByWebsocket(socket.id);
   const lobby = await getCurrentLobby(user.uuid);
 
   createGame(lobby.lobbyid, lobby.players);
 
+  addSubround(lobby.lobbyid);
+
   const gameData = startRound(1, 3);
   io.to(lobby.lobbyid).emit('startGame', gameData);
 };
 
-const cardPlayed = function (socket, io, payload) {
+const cardPlayed = async function (socket, io, payload) {
   console.log('Card played: ', payload);
 
-  // TODO: Play a card
+  const player = await getByWebsocket(socket.id);
+  const lobby = await getCurrentLobby(player.uuid);
+  const lobbyId = lobby.lobbyid;
+
+  try {
+    const { value, suit, trump } = payload;
+    const card = new Card(suit, value);
+
+    addCardPlayed(lobbyId, player, card);
+    payload.player = player.username;
+    io.to(lobbyId).emit('cardPlayed', payload);
+
+    // If all cards are played calculate outcome
+    const players = Object.keys(getPlayers(lobbyId));
+    const subround = getRounds(lobbyId)[getRounds(lobbyId).length - 1].subrounds[
+      getRounds(lobbyId)[getRounds(lobbyId).length - 1].subrounds.length - 1
+    ];
+
+    // Calculate winner
+    const cards = subround.cardsPlayed.map((play) => play.card);
+    const winningCard = getWinningCard(cards, trump);
+    const winner = subround.cardsPlayed.find((play) => play.card === winningCard);
+    setStichPlayer(lobbyId, winner.player);
```

Comment:
Doesn't this get executed even if not all cards are played?

Reply:
Yes, it executes after every played card. If we have enough time in Sprint 3 I would like to send the current winner of the subround to the FE and show it in the UI